### PR TITLE
XP-265 move benchmarks to separate repo

### PR DESCRIPTION
**Pull request by PanicButtonPressed**
### References

[XP-265](https://xainag.atlassian.net/browse/XP-265), [XP-243](https://xainag.atlassian.net/browse/XP-243)

### Summary

This PR solves the following:

- move benchmarks, examples and benchmark participant and coordinator out of the repository
- remove benchmark, ml training related dependencies and entrypoints from `setup.py`
- remove config files related to benchmarks (FLAGS, S3, AWS, Docker) and dataset pytest fixtures
- clean up the custom types
- remove `flags` from the codebase

The following preassumptions were made:

- docs are not adjusted and handled in another ticket
- commented out some logic of the `aggregator`, that has dependencies to the `evaluator` moved to `xain-benchmark` repository
- aggregation and selection (controller) logic will stay in the `xain-fl` repository at the moment since these are used by communication participant and coordinator
- some custom types are still used by the old communication participant

### Are there any open tasks/blockers for the ticket?

There are no blockers however, it has to be discussed how to continue with the `EvoAgg` and the `Evaluator` in general. Additionally, the docs have to be adjusted.

### Next Step

Modify docs to either reference to the `xain-benchmark` repository or create a doc for the benchmarking as well.
Furthermore, `flags` have to be replaced in `xain-benchmark`.

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [ ] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [ ] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -143,30 +129,14 @@ def run(self):
...
     cmdclass={"develop": CustomDevelopCommand},
-    entry_points={
-        "console_scripts": [
-            "coordinator=xain_fl.cproto.cli:main",
-            "train_remote=benchmarks.train_remote:main",
-            "pull_results=xain_fl.ops.__main__:download",
-            "aggregate=benchmarks.aggregate:main",
-        ]
-    },
+    entry_points={"console_scripts": ["coordinator=xain_fl.cproto.cli:main"]},
```
Comment (wilk10) [inline setup.py]: i think it's my fault, it's `xain_fl.cli:main` now (please double-check)

Comment (PanicButtonPressed) [reply setup.py]: Yeah, also my fault. I saw it and forgot to change it. Resolved.

File: CODEOWNERS [hunk 2]
```diff
@@ -6,20 +6,11 @@
 ## CI config
 /.circleci/ @r-marques @tanertopal
```
Comment (atymoshchuk) [inline CODEOWNERS]: We have to remove this as well, it's obsolete

Comment (PanicButtonPressed) [reply CODEOWNERS]: Thanks for pointing it out. This was done in another PR #195.

File: xain_fl/fl/coordinator/aggregate.py [hunk 3]
```diff
@@ -99,44 +85,56 @@ def federated_averaging(thetas: List[Theta], weighting: np.ndarray) -> Theta:
     return theta_avg
 
 
-def evo_agg(thetas: List[Theta], evaluator: Evaluator) -> Theta:
-    """Experimental
+# TODO: decide how to continue with that
```
Comment (atymoshchuk) [inline xain_fl/fl/coordinator/aggregate.py]: Please create a ticket for it and put a ticket number into todo

Comment (PanicButtonPressed) [reply xain_fl/fl/coordinator/aggregate.py]: Done.

**Pull request merged**